hr_holidays_hour/models/hr_holidays.py

```diff
@@ -0,0 +1,256 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
+# Copyright 2019 Coop IT Easy SCRLfs
+#   - Vincent Van Rossem <vincent@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import datetime
+import logging
+
+from openerp import _, api, fields, models
+from openerp.osv import osv
+from openerp.exceptions import ValidationError
+from openerp.exceptions import Warning as UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class HrHolidays(osv.osv):
+    _inherit = "hr.holidays"
+
+    # workaround to override constraints
+
+    def _check_date(self, cr, uid, ids, context=None):
+        return True
+
+    _check_holidays = lambda self, cr, uid, ids, context=None: self.check_holidays(
+        cr, uid, ids, context=context
+    )
+
+    _constraints = [
+        (
+            _check_date,
+            "You can not have 2 leaves that overlaps on same day!",
+            ["date_from", "date_to"],
+        ),
+        (
+            _check_holidays,
+            "The number of remaining leaves is not sufficient for this leave type.\n"
+            "Please verify also the leaves waiting for validation.",
+            ["state", "number_of_days_temp"],
+        ),
+    ]
+
+    def check_holidays(self, cr, uid, ids, context=None):
+        return True
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    number_of_hours_temp = fields.Float(
+        string="Allocation in Hours",
+        digits=(2, 2),
+        readonly=True,
+        states={
+            "draft": [("readonly", False)],
+            "confirm": [("readonly", False)],
+        },
+    )
+    number_of_hours = fields.Float(
+        compute="_compute_number_of_hours", store=True
+    )
+    virtual_hours = fields.Float(
+        compute="_compute_number_of_hours", store=True
+    )
+    working_hours = fields.Float(digits=(2, 2))
+
+    @api.onchange("employee_id")
+    def onchange_employee(self):
+        # Get result of original onchange_employee from parent class:
```

Comment:
When code is readable, no need for comments :-)

Reply:
👍